ansible_ai_connect_admin_portal/src/ModelSettingsKeyDeletionModal.tsx

```diff
@@ -0,0 +1,88 @@
+import { useTranslation } from "react-i18next";
+import { Button, Modal, ModalVariant, Text } from "@patternfly/react-core";
+import { useState } from "react";
+import { DELAY } from "./api/globals";
+import { APIException } from "./api/types";
+import { deleteWcaKey } from "./api/api";
+import { AxiosError } from "axios";
+import { HasError, NO_ERROR } from "./ErrorModal";
+import { BusyButton } from "./BusyButton";
+
+interface ModelSettingsKeyDeletionModalProps {
+  readonly isModalOpen: boolean;
+  readonly setIsModalOpen: (isOpen: boolean) => void;
+  readonly handleModalToggle: () => void;
+  readonly reloadParent: () => void;
+}
+
+export const ModelSettingsKeyDeletionModal = (
+  props: ModelSettingsKeyDeletionModalProps,
+) => {
+  const { t } = useTranslation();
+  const { isModalOpen, setIsModalOpen, handleModalToggle, reloadParent } =
+    props;
+  const [isDeleting, setDeleting] = useState<boolean>(false);
+  const [keyError, setKeyError] = useState<HasError>(NO_ERROR);
+
+  const deleteKey = () => {
+    const timeoutId = setTimeout(() => setDeleting(true), DELAY);
+    deleteWcaKey()
+      .then((_) => {
+        reloadParent();
+        handleCancel();
+      })
+      .catch((error: AxiosError<APIException, any>) => {
+        setKeyError({
+          inError: true,
+          message: error.message,
+          detail: error.response?.data?.detail ?? "",
+        });
+      })
+      .finally(() => {
+        setDeleting(false);
+        clearTimeout(timeoutId);
+      });
+  };
+
+  const handleCancel = () => {
+    setKeyError({
+      inError: false,
+    });
+    setIsModalOpen(false);
+  };
+
+  return (
+    <Modal
+      variant={ModalVariant.small}
+      title="Delete Confirmation"
+      isOpen={isModalOpen}
+      onClose={handleModalToggle}
+      data-testid={"model-settings-key-deletion-modal"}
+      actions={[
+        <BusyButton
+          key="delete"
+          variant="danger"
+          onClick={deleteKey}
+          isBusy={isDeleting}
```

Comment:
I use `isDisabled(..)` [elsewhere](https://github.com/ansible/ansible-ai-connect-service/pull/1333/files#diff-f227679fecf0d924a50722bc5f16ea60d0c7bccb3917df2732742d868883e249R285).

The button remains enabled whilst deleting, but does seems to prevent "double clicking" (whilst deleting).

For consistency with the other buttons I think disabling is better than marking it as "busy".